File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < n; i++) {
            map.put(i + 1, 1);
        }
        
        for (int lostNums : lost) {
            map.put(lostNums, map.get(lostNums) - 1);
        }
        
        for (int reserveNums : reserve ) {
            map.put(reserveNums, map.get(reserveNums) + 1);
        }
        
        for (int i = 1; i <= n; i++) {
            if (map.get(i) == 2) {
                if (i - 1 >= 1 && map.get(i - 1) == 0) {
                    map.put(i - 1, 1);
                    map.put(i, 1);    
                } 
                else if (i + 1 <= n && map.get(i + 1) == 0) {
                    map.put(i + 1, 1); 
                    map.put(i, 1);    
                }
            }
        }
        
        int answer = 0;
        for (int i = 1; i <= n; i++) {
            if (map.get(i) >= 1) {
                answer++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.cpp
#include <vector>
#include <unordered_set>
#include <algorithm>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    unordered_set<int> reserve_set(reserve.begin(), reserve.end());

    vector<int> real_lost;
    for (int l : lost) {
        if (reserve_set.count(l)) {
            reserve_set.erase(l);
        } else {
            real_lost.push_back(l);
        }
    }

    sort(real_lost.begin(), real_lost.end());

    for (int l : real_lost) {
        if (reserve_set.count(l - 1)) {
            reserve_set.erase(l - 1);
        } else if (reserve_set.count(l + 1)) {
            reserve_set.erase(l + 1);
        } else {
            n--;
        }
    }
    return n;
}